com.unity.render-pipelines.core/Tests/Editor/Volumes/VolumeComponentTests.cs

```diff
@@ -204,42 +205,101 @@ static private void TestAnimationCurveInterp(AnimationCurve lhsCurve, AnimationC
-            testCurve0.AddKey(new Keyframe(7.0f, 2.6f, -1.0f, -1.0f));
-
-            AnimationCurve testCurve1 = new AnimationCurve();
-            testCurve1.AddKey(new Keyframe(-1.0f, 3.0f, 2.0f, 2.0f));
-            testCurve1.AddKey(new Keyframe(4.0f, 2.0f, 3.0f, 3.0f));
-            testCurve1.AddKey(new Keyframe(5.0f, 2.6f, 0.0f, 0.0f));
-            testCurve1.AddKey(new Keyframe(9.0f, 2.6f, -5.0f, -5.0f));
-
-            // Test the same positions as curve 0 but different values and tangents
-            AnimationCurve testCurve2 = new AnimationCurve();
-            testCurve2.AddKey(new Keyframe(0.0f, 1.0f, -1.0f, 3.0f));
-            testCurve2.AddKey(new Keyframe(4.0f, 6.0f, -9.0f, -2.0f));
-            testCurve2.AddKey(new Keyframe(7.0f, 5.2f, -3.0f, -4.0f));
-
-            // Test the case where two curves have no overlap
-            AnimationCurve testCurve3 = new AnimationCurve();
-            testCurve3.AddKey(new Keyframe(11.0f, 1.0f, -1.0f, 3.0f));
-            testCurve3.AddKey(new Keyframe(14.0f, 6.0f, -9.0f, -2.0f));
-            testCurve3.AddKey(new Keyframe(17.0f, 5.2f, -3.0f, -4.0f));
-
-            TestAnimationCurveInterp(testCurve0, testCurve1, 0.5f, -2.0f, 10.0f, 100, 1e-5f, false);
-            TestAnimationCurveInterp(testCurve1, testCurve2, 0.5f, -2.0f, 10.0f, 100, 1e-5f, false);
-            TestAnimationCurveInterp(testCurve0, testCurve2, 0.5f, -2.0f, 10.0f, 100, 1e-5f, false);
-            TestAnimationCurveInterp(testCurve0, testCurve3, 0.5f, -2.0f, 25.0f, 100, 1e-5f, false);
+            AnimationCurve testCurve = new AnimationCurve();
+            if (index == 0)
+            {
+                testCurve.AddKey(new Keyframe(0.0f, 3.0f, 2.0f, 2.0f));
+                testCurve.AddKey(new Keyframe(4.0f, 2.0f, -1.0f, -1.0f));
+                testCurve.AddKey(new Keyframe(7.0f, 2.6f, -1.0f, -1.0f));
+            }
+            else if (index == 1)
+            {
+                testCurve.AddKey(new Keyframe(-1.0f, 3.0f, 2.0f, 2.0f));
+                testCurve.AddKey(new Keyframe(4.0f, 2.0f, 3.0f, 3.0f));
+                testCurve.AddKey(new Keyframe(5.0f, 2.6f, 0.0f, 0.0f));
+                testCurve.AddKey(new Keyframe(9.0f, 2.6f, -5.0f, -5.0f));
+            }
+            else if (index == 2)
+            {
+                // Needed for the same positions as curve 0 but different values and tangents
+                testCurve.AddKey(new Keyframe(0.0f, 1.0f, -1.0f, 3.0f));
+                testCurve.AddKey(new Keyframe(4.0f, 6.0f, -9.0f, -2.0f));
+                testCurve.AddKey(new Keyframe(7.0f, 5.2f, -3.0f, -4.0f));
+            }
+            else
+            {
+                // Need for the test case where two curves have no overlap
+                testCurve.AddKey(new Keyframe(11.0f, 1.0f, -1.0f, 3.0f));
+                testCurve.AddKey(new Keyframe(14.0f, 6.0f, -9.0f, -2.0f));
+                testCurve.AddKey(new Keyframe(17.0f, 5.2f, -3.0f, -4.0f));
+            }
```

Comment:
Note: this could be set in s_AnimationCurveTestPair. You can have like:
```
            new AnimationCurveTestPair
            {
                lhsIndex = 0,
                rhsIndex = 1,
                t = 0.25f,
                testName = "CurveTest 1"
            },
```
should be 
```
            new AnimationCurveTestPair
            {
                lhsCurve = CreateTestCurve(0),
                rhsCurve = CreateTestCurve(1),
                t = 0.25f,
                testName = "CurveTest 1"
            },
```